packages/react-widgets/src/hooks/useWidgetFilterValues.js

```diff
@@ -0,0 +1,27 @@
+import { selectSourceById } from '@carto/react-redux/';
+import { useSelector } from 'react-redux';
+import useCustomCompareMemo from './useCustomCompareMemo';
+import { dequal as deepEqual } from 'dequal';
+
+/**
+ * Obtain widget's filter values.
+ *
+ * @param  {object} props
+ * @param  {string} props.dataSource - ID of the source to get the filters from.
+ * @param  {string} props.id - ID of the widget that applied the filter.
+ * @param  {string} props.column - name of column of this widget.
+ * @param  {string} props.type - type of filter
+ */
+export function useWidgetFilterValues({ dataSource, id, column, type }) {
+  const columnFilters = useSelector((state) => {
+    const dataSourceState = selectSourceById(state, dataSource);
+
+    const filter = dataSourceState?.filters?.[column]?.[type];
+    if (!filter || filter.owner !== id) {
+      return null;
+    }
+    return filter.values;
+  });
```

Comment:
useSelector re-rendered each time any part of the redux store change. Also, I think it isn't recommended to use so much logic inside of an useSelector.
I would do something similar to this (not tested):

```suggestion
  const { filters } = useSelector((state) => selectSourceById(state, dataSource) || {});
  
  return useMemo(() => {
    const filter = filters?.[column]?.[type];
    if (!filter || filter.owner !== id) {
      return null;
    }
    return filter.values;
  }, [filters, column, type, id]);
```

Reply:
Why not using a new createSelector for this, instead of reading + filtering an existing one?
From my understanding, using a createSelector should be enough to create a memoized selection, but we should check.

Pretty relevant links here: https://redux-toolkit.js.org/api/createSelector

Reply:
Yes, createSelector can be an option but from my understanding of createSelector it's for complex data structures and calculations using different parts of redux state.
I don't think it's the case where we just take filters from redux and the operations made (accessing a key-value object) have a insigficant impact. Even we can discuss if useMemo adds enough perf improvements that worth memoization cost haha.
I think useMemo is good enough.

Reply:
It's true that createSelector allows combining several slices for a query, but they are also used for basic / repeated stuff like filtering: https://github.com/reduxjs/reselect. IMO main point is 'encapsulation', not perf.

In any case, let's leave @zbigg pick an option here

Reply:
I think that @Clebal comments are ok, i've applied them in e391abd.